controllers/volumereplicationgroup_controller.go

```diff
@@ -652,16 +667,27 @@ func (v *VRGInstance) validatePVExistence(pv *corev1.PersistentVolume) error {
 		return fmt.Errorf("failed to get existing PV (%w)", err)
 	}
 
-	if existingPV.ObjectMeta.Annotations == nil ||
-		existingPV.ObjectMeta.Annotations[PVRestoreAnnotation] == "" {
-		return fmt.Errorf("found PV object not restored by Ramen for PV %s", existingPV.Name)
+	if existingPV.ObjectMeta.Annotations != nil &&
+		existingPV.ObjectMeta.Annotations[PVRestoreAnnotation] == "True" {
+		// Should we check and see if PV in being deleted? Should we just treat it as exists
+		// and then we don't care if deletion takes place later, which is what we do now?
+		v.log.Info("PV exists and managed by Ramen", "PV", existingPV)
+
+		return nil
 	}
 
-	// Should we check and see if PV in being deleted? Should we just treat it as exists
-	// and then we don't care if deletion takes place later, which is what we do now?
-	v.log.Info("PV exists and managed by Ramen", "PV", existingPV)
+	// Right now, we can only have either Sync mode or Async mode. In case of
+	// sync mode, the pv is never deleted as part of the failover/relocate
+	// process. Hence, the restore is not required and the annotation for
+	// restore can be missing for the sync mode. Skip the check for the
+	// annotation in this case.
+	if v.instance.Spec.Sync.Mode == ramendrv1alpha1.SyncModeEnabled {
```

Comment:
(nit) I think this block should be after line 617. It is misleading that the validation function ends up also updating the pv object.  IOW, if validation fails, then we update the PV object if the sync mode is enabled.

Reply:
This is a change we wanted to make too. However, validatePVExistence can return in two cases, PV is missing and PV annotations are missing. We thought it would be better to refactor the function later and it touches the async DR code too.